### PR TITLE
[Scala] add async support to Scala API client 

### PR DESCRIPTION
**Pull request by wing328**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [x] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

Based on #4855

cc @clasnake

### CODE REVIEW

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/ScalaClientCodegen.java [hunk 1]
```diff
@@ -75,8 +80,8 @@ public ScalaClientCodegen() {
         importMapping.remove("Set");
         importMapping.remove("Map");
 
-        importMapping.put("DateTime", "org.joda.time.DateTime");
```
Comment (wing328) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/ScalaClientCodegen.java]: @clasnake there's a PR by @epaul to add back the missing dependencies for joda time: https://github.com/swagger-api/swagger-codegen/pull/5092

If I remember correctly, we encountered issue with java.util.Date in the Java API client and therefore @cbornet introduced other datetime libraries (e.g. joda) to address the issue.

**Comment by wing328** [comment]
@clasnake CI (travis) reported errors. Here is part of the error messages:
```
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/client/petstore/scala/src/main/scala/io/swagger/client/AsyncClient.scala:5: error: object wordnik is not a member of package com
[ERROR] import com.wordnik.swagger.client._
[ERROR]            ^
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/client/petstore/scala/src/main/scala/io/swagger/client/AsyncClient.scala:9: error: not found: type SwaggerConfig
[ERROR] class AsyncClient(config: SwaggerConfig) extends Closeable {
[ERROR]                           ^
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/client/petstore/scala/src/main/scala/io/swagger/client/AsyncClient.scala:15: error: not found: type TransportClient
[ERROR]   protected def transportClient: TransportClient = new RestClient(config)
[ERROR]                                  ^
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/client/petstore/scala/src/main/scala/io/swagger/client/AsyncClient.scala:15: error: not found: type RestClient
[ERROR]   protected def transportClient: TransportClient = new RestClient(config)
[ERROR]                                                        ^
[ERROR] /home/travis/build/swagger-api/swagger-codegen/samples/client/petstore/scala/src/main/scala/io/swagger/client/api/PetApi.scala:32: error: object wordnik is not a member of package com
[ERROR] import com.wordnik.swagger.client._
[ERROR]            ^
```
Ref: https://s3.amazonaws.com/archive.travis-ci.org/jobs/212403027/log.txt

I wonder if you can take a look as I might have made mistakes when resolving conflicts during the cherry-pick.

**Comment by clasnake** [comment]
@wing328 No problem. I'll take a look. Do I have the permission to update this pr directly? Or shall I create a new one against this one or do you have any other suggestions?

**Comment by wing328** [comment]
@clasnake I don't think you've permission as the PR is in the official repo.

Please feel free to submit against this one (pr4855_cherrypick) or 2.3.0.

**Comment by wing328** [comment]
Closed via https://github.com/swagger-api/swagger-codegen/pull/5115

**Pull request closed without merging**